### PR TITLE
Fix GPy.priors.InverseGamma

### PR DESCRIPTION
**Pull request by ekalosak**
fixed InverseGamma prior: beforehand, it was a child class of Gamma but it defined a broken __new__ method of its own. Now, it just inherits Gamma's __new__; also added a test that ensures the InverseGamma can be instantiated and integrated into a GPy model

### CODE REVIEW

**Comment by ekalosak** [comment]
Also, I'm curious why GPy.core.Model is Priorizable and has a set_prior() method - it's not a parameter itself.

File: GPy/core/parameterization/priors.py [hunk 1]
```diff
@@ -357,24 +357,17 @@ class InverseGamma(Gamma):
     """
     domain = _POSITIVE
     _instances = []
-    def __new__(cls, a=1, b=.5): # Singleton:
```
Comment (lawrennd) [inline GPy/core/parameterization/priors.py:360]: Does removing this mean that those default settings are lost? Or are they the same as the original Gamma?

**Review by lawrennd — APPROVED** [review]
Looks fine to me. 

I haven't dug into the Gamma prior though and checked that it interprets a and b in the same way when constructing.

File: GPy/core/parameterization/priors.py (hunk 1 above)
Comment (ekalosak) [reply GPy/core/parameterization/priors.py:360]: They're identical to the defaults from `class Gamma(Prior)`'s `__new__` as defined on `GPy/core/parameterization/priors.py#L291` in `devel:HEAD` at time of writing (i.e. at commit `ff82f12c3d321bfc3ce6615447fad25aea9de6bd`).

**Pull request merged**